tgui/packages/tgui/interfaces/LocalGenerator/index.tsx

```diff
@@ -0,0 +1,186 @@
+
+  const handleToggleGenerator = () => {
+    act("toggle-generator");
+  };
+
+  const handleSwapChargeTarget = () => {
+    act("swap-target");
+  };
+
+  const handleEjectTank = () => {
+    act("eject-tank");
+  };
+
+  const handleEjectCell = () => {
+    act("eject-cell");
+  };
+
+  const handleConnectAPC = () => {
+    act("connect-APC");
+  };
+
+  return (
+    <Window title={name} width={300} height={470}>
+      <Window.Content>
+        <Stack vertical>
+          <Section title="Generator">
+            <Stack vertical>
+              <Stack.Item>
+                <Stack align="baseline">
+                  <Stack.Item basis="50%">
+                    <Button
+                      width={11}
+                      content="Toggle Floor Bolts"
+                      onClick={() => handleToggleBolts()} />
+                  </Stack.Item>
+                  <Stack.Item>
+                    Status:
+                  </Stack.Item>
+                  {boltsStatus ? (
+                    <Stack.Item color="good">
+                      ACTIVE
+                    </Stack.Item>
+                  ) : (
+                    <Stack.Item color="average">
+                      INACTIVE
+                    </Stack.Item>
+                  )}
+                </Stack>
+              </Stack.Item>
+              <Stack.Item>
+                <Stack align="baseline">
+                  <Stack.Item basis="50%">
+                    <Button
+                      width={11}
+                      content="Toggle Generator"
+                      onClick={() => handleToggleGenerator()} />
+                  </Stack.Item>
+                  <Stack.Item>
+                    Status:
+                  </Stack.Item>
+                  {generatorStatus ? (
+                    <Stack.Item color="good">
+                      ACTIVE
+                    </Stack.Item>
+                  ) : (
+                    <Stack.Item color="average">
+                      INACTIVE
+                    </Stack.Item>
+                  )}
+                </Stack>
+              </Stack.Item>
+              <Stack.Item>
+                <Stack align="baseline">
+                  <Stack.Item basis="50%">
+                    <Button
+                      width={11}
+                      content="Change Target"
+                      onClick={() => handleSwapChargeTarget()} />
+                  </Stack.Item>
+                  <Stack.Item>
+                    Target:
+                  </Stack.Item>
+                  {chargeAPC ? (
+                    <Stack.Item color="good">
+                      APC
+                    </Stack.Item>
+                  ) : (
+                    <Stack.Item color="good">
+                      CELL
+                    </Stack.Item>
+                  )}
+                </Stack>
+              </Stack.Item>
+            </Stack>
+          </Section>
+          <PortableHoldingTank
+            holding={holding}
```

Comment:
No objection to making the `holding` prop name here more meaningful if you have enough issues with the name to need a comment where the prop is destructured.